protocols/ping/src/protocol.rs

```diff
@@ -52,35 +49,6 @@ pub struct Ping;
 
 const PING_SIZE: usize = 32;
 
-impl UpgradeInfo for Ping {
-    type Info = &'static [u8];
-    type InfoIter = iter::Once<Self::Info>;
-
-    fn protocol_info(&self) -> Self::InfoIter {
-        iter::once(PROTOCOL_NAME)
-    }
-}
-
-impl InboundUpgrade<NegotiatedSubstream> for Ping {
-    type Output = NegotiatedSubstream;
-    type Error = Void;
-    type Future = future::Ready<Result<Self::Output, Self::Error>>;
-
-    fn upgrade_inbound(self, stream: NegotiatedSubstream, _: Self::Info) -> Self::Future {
-        future::ok(stream)
-    }
-}
-
-impl OutboundUpgrade<NegotiatedSubstream> for Ping {
-    type Output = NegotiatedSubstream;
-    type Error = Void;
-    type Future = future::Ready<Result<Self::Output, Self::Error>>;
-
-    fn upgrade_outbound(self, stream: NegotiatedSubstream, _: Self::Info) -> Self::Future {
-        future::ok(stream)
-    }
-}
-
```

Comment:
_Emphasizing this, though I think we are of the same opinion:_ While not de-duplicating given that `ReadyUpgrade` is only used once, I do think these kind of pull requests do a great job simplifying the business logic, here `libp2p-ping`, moving auxiliary not-directly-related logic to common places, here `libp2p-core`.

Reply:
Yes, strong agreement here.